experimental/BasisLieHighestWeight/test/MainAlgorithm-test.jl

```diff
@@ -609,3 +609,83 @@ end
+      
+      type, rank, highest_weight = case
+      weyl_group_elem = convert(Vector{Int64}, word(longest_element(weyl_group(type, rank))))
+
+      # longest weyl group element leads to the same result as a simple module
+      mb1 = basis_lie_highest_weight(type, rank, highest_weight; monomial_ordering)
+      mb2 = basis_lie_demazure(type, rank, highest_weight, weyl_group_elem; monomial_ordering)
+      @test monomials(mb1) == monomials(mb2)
+
+      # empty weyl group element leads to a monomialbasis with only the one element
+      mb = basis_lie_demazure(type, rank, highest_weight, Int[]; monomial_ordering)
+      @test monomials(mb) == Set{ZZMPolyRingElem}([one(mb.monomials_parent)])
+    end
+  end
+
+  @testset "Nontrivial Cases" begin
+    for case in [
+      (:A, 3, [1, 0, 1], [1, 2, 1]),
+      (:A, 3, [2, 2, 2], [1, 2, 1]),
+      (:A, 3, [2, 1, 0], [1, 2, 1]), #example below
+      #(:B, 2, [1, 0], [1, 2]),
+      #(:C, 2, [1, 1], [1, 2, 1]),
+      (:D, 4, [1, 0, 1, 0], [1, 2, 1]),
+      #(:G, 2, [1, 0], [1, 2, 1]),
+    ], monomial_ordering in [:degrevlex, :lex, :invlex, :neglex, :wdegrevlex]
+
+      type, rank, highest_weight, weyl_group_elem = case
+
+      mb = basis_lie_demazure(type, rank, highest_weight, weyl_group_elem; monomial_ordering)
+      R = root_system(birational_sequence(mb))
+
+      char = demazure_character(R, highest_weight, weyl_group_elem)
+      
+      dict = Dict{WeightLatticeElem,Int64}()
+      for mon in monomials(mb)
+        w = WeightLatticeElem(R, highest_weight) - Oscar.BasisLieHighestWeight.weight(mon, birational_sequence(mb))
+        val = get(dict, w, 0) + 1
+        dict[w] = val
+      end
+      @test dict == char
+    end
+  end
+  @testset "Hand Computed Case" begin
+    # Test for a specific case in which dependent on the order there are two out of three monomials in a weight space
+    type = :A
+    rank = 3
+    highest_weight = [2, 1, 0]
+    weyl_group_elem = [1, 2, 1]   
+
+    for case in [
+      (:degrevlex, Set{Vector{Int64}}([[0, 0, 2], [1, 1, 1],])),
+      (:lex, Set{Vector{Int64}}([[0, 0, 2], [1, 1, 1],])),
+      #(:invlex, Set{Vector{Int64}}([[2, 2, 0], [1, 1, 1],])),
+      #(:neglex, Set{Vector{Int64}}([[2, 2, 0], [1, 1, 1],])),
+      (:wdegrevlex, Set{Vector{Int64}}([[0, 0, 2], [1, 1, 1],])),
+    ]
+      monomial_ordering, expected_result = case
+      mb = basis_lie_demazure(type, rank, highest_weight, weyl_group_elem; monomial_ordering)
+    
+      R = root_system(birational_sequence(mb))
+      w = WeightLatticeElem(R, highest_weight) - 2 * WeightLatticeElem(simple_root(R, 1)) - 2 * WeightLatticeElem(simple_root(R, 2))
+
+      monomials_for_weight_w = Set{ZZMPolyRingElem}([mon for mon in monomials(mb) if WeightLatticeElem(R, highest_weight) - Oscar.BasisLieHighestWeight.weight(mon, birational_sequence(mb)) == w])
+
+      exponent_vectors = Set{Vector{Int64}}([collect(exponents(m))[1] for m in monomials_for_weight_w])
```

Comment:
better use one of the following two options
```suggestion
      exponent_vectors = Set{Vector{Int64}}([only(exponents(m)) for m in monomials_for_weight_w])
```
```suggestion
      exponent_vectors = Set{Vector{Int64}}([leading_exponent(m) for m in monomials_for_weight_w])
```